packages/node/test/index.test.ts

```diff
@@ -192,6 +193,123 @@ describe('#metrics', function () {
     });
   });
 
+  describe('unified snippet tests', function () {
+    let metricsServerRequests: number;
+    let app: Express;
+    let metricsServerResponseCode: number;
+
+    beforeEach(function () {
+      metricsServerRequests = 0;
+      metricsServerResponseCode = 202;
+      server.use(
+        ...[
+          rest.post(`${config.host}/v1/request`, async (req, res, ctx) => {
+            const body: OutgoingLogBody[] = await req.json();
+            if (doMetricsHeadersMatch(req.headers)) {
+              metricsServerRequests += 1;
+              expect(body[0]._version).to.equal(3);
+              expect(body[0].group).to.deep.equal(outgoingGroup);
+              expect(typeof body[0].request.log.entries[0].startedDateTime).to.equal('string');
+              return res(ctx.status(metricsServerResponseCode), ctx.text(''));
+            }
+
+            return res(ctx.status(500));
+          }),
+          rest.get(`${config.readmeApiUrl}/v1`, (req, res, ctx) => {
+            return res(
+              ctx.status(200),
+              ctx.json({
+                jwtSecret: '123',
+                subdomain: 'subdomain',
+              }),
+            );
+          }),
+          rest.get(`${config.readmeApiUrl}/v1/version`, (req, res, ctx) => {
+            return res(
+              ctx.status(200),
+              ctx.json([
+                {
+                  version: '1.0',
+                  subdomain: 'subdomain',
+                },
+              ]),
+            );
+          }),
+        ],
+      );
+
+      readmeio.auth(apiKey);
+      app = express();
+      app.use(
+        readmeio.readme((req, getUser) => {
+          return getUser({
+            byAPIKey: (requestApiKey: string) => {
+              // TODO should we be calling this if the requestApiKey is undefined?
+              if (!requestApiKey) {
+                return undefined;
+              }
+
+              return {
+                keys: [{ apiKey: requestApiKey, name: 'test' }],
+                name: 'test',
+                email: 'test@example.com',
+              };
+            },
+            byEmail: (email: string) => {
+              if (!email) {
+                return undefined;
+              }
+
+              return {
+                keys: [{ apiKey: endUserApiKey, name: 'test' }],
+                name: 'test',
+                email: 'test@example.com',
+              };
+            },
+          });
+        }),
+      );
+      app.get('/test', (req, res) => {
+        return res.sendStatus(200);
+      });
+    });
+
+    afterEach(function () {
+      setBackoff(undefined);
+      metricsServerResponseCode = 202;
+    });
+
+    function makeRequest(query = '') {
+      return request(app).get(`/test${query}`).expect(200);
+    }
+
+    it('should send requests to the metrics server', async function () {
+      expect.assertions(10);
```

Comment:
nit: the `expect.assertions` is a little confusing. Could we sub it with [toHaveBeenCalledTimes](https://jestjs.io/docs/expect#tohavebeencalledtimesnumber)?

Reply:
cc @kanadgupta

Reply:
mock functions aren't applicable here, `expect.assertions` ensures that all the `expect` statements over here are being hit properly:

https://github.com/readmeio/metrics-sdks/blob/dd18d505fdda8aa82580fe729b5a6af6c6cf0628/packages/node/test/index.test.ts#L139-L141

this pattern isn't great and i'm curious if others that have been in the weeds with MSW have better ideas for asserting request header content (or if that's even necessary) // @erunion

Reply:
doing `expect()` in here or having the MSW mock return a 500 status code with a specific error is probably the best way to handle these